tools/samtools.wdl

```diff
@@ -102,12 +107,34 @@ task split {
             ~{prefix}.unaccounted_reads.bam \
             > first_unaccounted_read.sam
 
-        if ~{reject_unaccounted} && [ -s first_unaccounted_read.sam ]; then
-            exit 42
+        if ~{reject_unaccounted_reads} && [ -s first_unaccounted_read.sam ]; then
+            >&2 echo "There are reads present with bad or missing RG tags!"
+            exit 21
         else
             rm ~{prefix}.unaccounted_reads.bam
         fi
         rm first_unaccounted_read.sam
+
+        # Check that all output BAMs have at least
+        # one read in them.
+        if ~{reject_empty_output}; then
+            for out_bam in *.bam; do
+                samtools head \
+                    --threads "$n_cores" \
+                    -h 0 \
+                    -n 1 \
+                    "$out_bam" \
+                    > first_read.sam
+
+                if [ ! -s first_read.sam ]; then
+                    >&2 echo "No reads are in output BAM $out_bam!"
+                    >&2 echo "This is likely caused by malformed RG records."
+                    rm first_read.sam
+                    exit 42
```

Comment:
Can you modify this to wait and exit after checking all BAMs? I doubt there would be multiple empty read groups, but we should probably catch that case in a single check.

Reply:
Good idea